### PR TITLE
feat(compaction): limit write-amplification for TierCompactionPicker 

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- reduce repeated code
- limit write-amplification for L0 compaction. For example, if some file merged by overlapping level (which are generated by commit epoch) are 1MB, we limit at least 4 times to merge to a large one, so it will be merge at most 4 times (1MB --> 4MB --> 16MB --> 64MB --> 256MB, and 256MB is the largest size of non-overlapping sub-level in L0), the write-amplification is 4. And if the files generated by commit epoch are larger ,we could get a much smaller write-amplification.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by Little-Wallace** [comment]
Main branch:
image

Current branch:
image

File: src/meta/src/hummock/compaction/tier_compaction_picker.rs [hunk 1]
```diff
@@ -39,145 +39,78 @@ impl TierCompactionPicker {
...
-                .sum::<usize>()
-                < self.config.level0_tier_compact_file_number as usize
-            {
+            if compact_file_count < self.config.level0_tier_compact_file_number as usize {
                 continue;
             }
 
-            if select_level_inputs.len() <= self.config.level0_tier_compact_file_number as usize / 2
-            {
+            let is_write_amp_large =
```
Comment (Li0k) [inline src/meta/src/hummock/compaction/tier_compaction_picker.rs:104]: Could you add some comments to describe how to calculate is_write_amp_large and why the l0 file_number need to divide 2 ?

**Review by Li0k — APPROVED** [review]
Rest LGTM

File: src/meta/src/hummock/compaction/tier_compaction_picker.rs (hunk 1 above)
Comment (Little-Wallace) [reply src/meta/src/hummock/compaction/tier_compaction_picker.rs:104]: OK.

**Pull request merged**